### PR TITLE
New metric: Fowlkes-Mallows Index

### PR DESCRIPTION
**Pull request by matsumotosan**
## What does this PR do?

Fixes part of #2003

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/functional/clustering/fowlkes_mallows_index.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        contingency: contingency matrix
+        n: number of samples
+
+    Returns:
+        fowlkes_mallows: Fowlkes-Mallows index
+
+    """
+    tk = torch.sum(contingency**2) - n
+    if torch.allclose(tk, tensor(0)):
+        return torch.tensor(0.0)
```
Comment (SkafteNicki) [inline src/torchmetrics/functional/clustering/fowlkes_mallows_index.py:50]:
```suggestion
        return torch.tensor(0.0, device=contingency.device)
```

**Review by SkafteNicki — APPROVED** [review]

**Review by Borda — APPROVED** [review]
